mutation/mutation.cc

```diff
@@ -194,6 +196,71 @@ mutation reverse(mutation mut) {
     return *std::move(mut).consume(reverse_rebuilder, consume_in_reverse::yes).result;
 }
 
+future<> split_mutation(mutation source, std::vector<mutation>& target, size_t max_size) {
+    class splitter {
+        const schema_ptr _schema;
+        std::vector<mutation>& _target;
+        std::optional<mutation_rebuilder_v2> _mut_builder;
+        size_t _size = 0;
+        const size_t _max_size;
+        void maybe_flush() {
+            if (_size >= _max_size) {
```

Comment:
Shouldn't we reset _size back to 0 after flushing?
I think this will cause us to start flushing after every row once we exceed _max_size for the first time.